pg_search/src/query/mod.rs

```diff
@@ -569,8 +569,19 @@ impl SearchQueryInput {
             return None;
         }
 
-        if let Some(array_query) = Self::coerce_scalar_array(datum, is_null) {
-            return Some(array_query);
+        // Check if this is a SearchQueryInput array (ScalarArrayOpExpr case).
+        // Without this check, FromDatum would panic with "cache lookup failed" on non-array datums.
+        let array = datum.cast_mut_ptr::<pg_sys::ArrayType>();
+        let is_sqi_array = (*array).ndim >= 1
+            && (*array).ndim <= pg_sys::MAXDIM as i32
+            && (*array).elemtype == searchqueryinput_typoid();
```

Comment:
Aha, that makes sense. I suspected at least some of the handling going on here was strictly necessary, just not all of them.